File: LeetCode_Marathons/LCM6/Easy/Detect_Capital/a.py
class Solution:
    def isMonotic(self, word: str, isUpper: bool) -> bool:
        for letter in word:
            if isUpper and not ("A" <= letter and letter <= "Z"):
                return False
            if not isUpper and not ("a" <= letter and letter <= "z"):
                return False
        return True
    

    def detectCapitalUse(self, word: str) -> bool:
        if "a" <= word[0] and word[0] <= "z":
            return self.isMonotic(word[1:], False)
        return self.isMonotic(word[1:], False) or self.isMonotic(word[1:], True)

File: LeetCode_Marathons/LCM6/Easy/Detect_Capital/a.cpp
class Solution {
private:
    bool isMonotic(string word, bool isUpper) {
        for (char letter : word) {
            if (isUpper && !('A' <= letter && letter <= 'Z')) {
                return false;
            }
            if (!isUpper && !('a' <= letter && letter <= 'z')) {
                return false;
            }
        }

        return true;
    }
public:
    bool detectCapitalUse(string word) {
        if ('a' <= word[0] && word[0] <= 'z') {
            return isMonotic(word.substr(1), false);
        }
        return isMonotic(word.substr(1), false) || isMonotic(word.substr(1), true);
    }
};